
# HG changeset patch
# User Ori Bernstein <ori@eigenstate.org>'
# Date 1588022377 25200
# Node ID 8fd96772eb02c2f55d895d4a125a3826f4b344ab
# Parent  4911d154d158ff29091b5cc2e752838dc3258991
fix drawterm compiles using gcc 10

-fno-common became the default, and and kmesg was relying
on common symbols being merged.

diff -r 4911d154d158 -r 8fd96772eb02 kern/dat.h
--- a/kern/dat.h	Thu Apr 23 12:52:37 2020 -0700
+++ b/kern/dat.h	Mon Apr 27 14:19:37 2020 -0700
@@ -32,6 +32,7 @@
 typedef struct RWlock	RWlock;
 typedef struct Waitq	Waitq;
 typedef struct Walkqid	Walkqid;
+typedef struct Kmesg	Kmesg;
 typedef int    Devgen(Chan*, char*, Dirtab*, int, int, Dir*);
 
 #include "fcall.h"
@@ -477,8 +478,10 @@
  * Log console output so it can be retrieved via /dev/kmesg.
  * This is good for catching boot-time messages after the fact.
  */
-struct {
+struct Kmesg {
 	Lock lk;
 	uint n;
 	char buf[16384];
-} kmesg;
+};
+
+extern Kmesg kmesg;
diff -r 4911d154d158 -r 8fd96772eb02 kern/devcons.c
--- a/kern/devcons.c	Thu Apr 23 12:52:37 2020 -0700
+++ b/kern/devcons.c	Mon Apr 27 14:19:37 2020 -0700
@@ -13,6 +13,7 @@
 
 void	(*screenputs)(char*, int) = 0;
 
+Kmesg	kmesg;			/* console messages */
 Queue*	kbdq;			/* unprocessed console input */
 Queue*	lineq;			/* processed console input */
 Queue*	kprintoq;		/* console output, for /dev/kprint */

